### PR TITLE
Refactor publication edit additional info

### LINKED ISSUE #1530: publication additional information edit

### PR DESCRIPTION
**Pull request by nics**
fixes #1530
depends on #1529

### CODE REVIEW

File: handlers/publicationediting/abstract.go [hunk 1]
```diff
@@ -29,163 +26,148 @@ type BindDeleteAbstract struct {
...
-func (h *Handler) CreateAbstract(w http.ResponseWriter, r *http.Request, ctx Context) {
+func CreateAbstract(w http.ResponseWriter, r *http.Request) {
+	c := ctx.Get(r)
+	p := ctx.GetPublication(r)
+
 	b := BindAbstract{}
 	if err := bind.Request(r, &b, bind.Vacuum); err != nil {
-		h.Logger.Warnw("create publication abstract: could not bind request arguments", "errors", err, "request", r, "user", ctx.User.ID)
+		c.Log.Warnw("create publication abstract: could not bind request arguments", "errors", err, "request", r, "user", c.User.ID)
 		render.BadRequest(w, r, err)
```
Comment (verheyenkoen) [inline handlers/publicationediting/abstract.go:43]: To convert (`c.HandleError(...)`)

File: handlers/publicationediting/abstract.go [hunk 2]
```diff
@@ -29,163 +26,148 @@ type BindDeleteAbstract struct {
...
-			Conflict: true,
-		})
+		views.ReplaceModal(publicationviews.EditAbstractDialog(c, p, abstract, idx, true, nil, true)).Render(r.Context(), w)
 		return
 	}
 
 	if err != nil {
-		h.Logger.Errorf("create publication abstract: could not save the publication:", "errors", err, "publication", ctx.Publication.ID, "user", ctx.User.ID)
+		c.Log.Errorf("create publication abstract: could not save the publication:", "errors", err, "publication", p.ID, "user", c.User.ID)
 		render.InternalServerError(w, r, err)
```
Comment (verheyenkoen) [inline handlers/publicationediting/abstract.go:76]: To convert (`c.HandleError(...)`)

File: views/publication/edit_abstract.templ [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	<div class="modal-dialog modal-dialog-centered modal-lg modal-dialog-scrollable" role="document">
+		<div class="modal-content">
+			<div class="modal-header">
+				<h2 class="modal-title">
+					if isNew {
+						Add abstract
+					} else {
+						Edit abstract
+					}
+					x
```
Comment (verheyenkoen) [inline views/publication/edit_abstract.templ:22]: What is this "x"?

**Review by verheyenkoen — CHANGES_REQUESTED** [review]

File: handlers/publicationediting/abstract.go (hunk 2 above)
Comment (nics) [reply handlers/publicationediting/abstract.go]: fixed

File: handlers/publicationediting/abstract.go (hunk 1 above)
Comment (nics) [reply handlers/publicationediting/abstract.go]: fixed

**Review by verheyenkoen — APPROVED** [review]

**Pull request merged**